core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -548,23 +542,22 @@ class GroupMetadataManager(val brokerId: Int,
     cleanupGroupMetadata(None)
   }
 
-  def cleanupGroupMetadata(topicPartitions: Option[Seq[TopicPartition]]) {
+  def cleanupGroupMetadata(deletedTopicPartitions: Option[Seq[TopicPartition]]) {
     val startMs = time.milliseconds()
     var offsetsRemoved = 0
 
     groupMetadataCache.foreach { case (groupId, group) =>
-      val (expiredOffsets, groupIsDead, generation) = group synchronized {
-        // remove expired offsets from the cache
-        val expiredOffsets = if (topicPartitions.isEmpty)
-          group.removeExpiredOffsets(startMs)
-        else
-          group.removeOffsets(topicPartitions.get)
+      val (removedOffsets, groupIsDead, generation) = group synchronized {
+        val removedOffsets = deletedTopicPartitions match {
+          case Some(topicPartitions) => group.removeOffsets(topicPartitions)
```

Comment:
might be overkill, but I also considered passing in a function `GroupMetadata => Map[TopicPartition, OffsetAndMetadata]` as an argument to alleviate some of the awkwardness of handling both cases here.